jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -371,37 +371,17 @@ def _nddata_to_glue_data(ndd, data_label):
     if ndd.data.ndim != 2:
         raise ValueError(f'Imviz cannot load this NDData with ndim={ndd.data.ndim}')
 
-    for attrib, sub_attrib in zip(
-            ['data', 'mask', 'uncertainty'],
-            [None, None, 'array']
-    ):
+    for attrib in ('data', 'mask', 'uncertainty'):
         arr = getattr(ndd, attrib)
         if arr is None:
             continue
-        cur_data = Data()
+        comp_label = attrib.upper()
+        cur_label = f'{data_label}[{comp_label}]'
+        cur_data = Data(label=cur_label)
         cur_data.meta.update(standardize_metadata(ndd.meta))
         if ndd.wcs is not None:
             cur_data.coords = ndd.wcs
         raw_arr = arr
-
-        if sub_attrib is not None:
-            # since NDDataArray.uncertainty may be an object like
-            # StdDevUncertainty, we need to take another attr
-            # like StdDevUncertainty.array:
-            base_arr = getattr(raw_arr, sub_attrib)
-        else:
-            base_arr = raw_arr
-        wcs_only = np.all(np.isnan(base_arr))
-
-        if 'WCS-ONLY' not in cur_data.meta or not cur_data.meta.get('WCS-ONLY'):
-            cur_data.meta.update({'WCS-ONLY': wcs_only})
-
-        cur_label = f'{data_label}'
-        comp_label = attrib.upper()
-        if not wcs_only:
-            cur_label += f'[{comp_label}]'
-        cur_data.label = cur_label
```

Comment:
I think you reverted all changes in the parser in your PR, but this particular one was useful because the default labeling behavior appends `[DATA]` to each WCS-only Data label. Otherwise you have this:

<img width="467" alt="Screen Shot 2023-05-18 at 09 36 26" src="https://github.com/bmorris3/jdaviz/assets/3497584/0f1c36ed-81e8-4d79-b815-b1bc01c47a51">

That `[DATA]` is useful when there is ambiguity about which attributes were parsed, but there is no ambiguity in this case. Can we go back to this behavior?

Reply:
Can we have a special hidden function for parsing WCS-only stuff then? I still don't like injecting code like this in the user-facing parser. What if we somehow decide not to load them as NDData in the future, or user somehow go through this logic route for their own data and get inconsistent labeling behavior? I'll think about your comment and propose something in my follow up commit(s).

Reply:
Would the new `_wcsonly2data` function be sufficient for you? It returns the `Data` that you can add to the app and the viewer.